src/backend/distributed/commands/view.c

```diff
@@ -35,11 +35,44 @@
 #include "utils/lsyscache.h"
 #include "utils/syscache.h"
 
+/*
+ * GUC stops distributing local views that has no distributed dependency
+ * to prevent circular dependency error logs. It is intended to be used
+ * before postgres vanilla tests.
+ */
+bool DistributeLocalViews = true;
+
 static List * FilterNameListForDistributedViews(List *viewNamesList, bool missing_ok);
 static void AppendQualifiedViewNameToCreateViewCommand(StringInfo buf, Oid viewOid);
 static void AppendViewDefinitionToCreateViewCommand(StringInfo buf, Oid viewOid);
 static void AppendAliasesToCreateViewCommand(StringInfo createViewCommand, Oid viewOid);
 static void AppendOptionsToCreateViewCommand(StringInfo createViewCommand, Oid viewOid);
+static bool ViewHasDistributedDependency(ObjectAddress *viewObjectAddress);
+
+/*
+ * ViewHasDistributedDependency returns true if given view at address has
+ * any distributed dependency.
+ */
+static bool
+ViewHasDistributedDependency(ObjectAddress *viewObjectAddress)
+{
+	bool hasDistDependency = false;
+	List *dependencies = GetAllDependenciesForObject(viewObjectAddress);
```

Comment:
The problem with this approach is that the dependency schema is could be distributed.

I guess we should re-write this to `ViewHasDistributedRelationDependency` such that we only check `IsAnyObjectDistributed` for `dependency.classId = RelationRelationId` 

Do you see any problems with that? 


In the end, we should make sure that the following views can be created with the guc disabled:
```SQL
CREATE TABLE pg_local (a int);
CREATE VIEW v3 as SELECT * FROM pg_local;
CREATE VIEW v2 as SELECT * FROM pg_local;
CREATE OR REPLACE VIEW v3 AS SELECT * FROM v2;
CREATE OR REPLACE VIEW v2 AS SELECT * FROM v3;
```